src/clients/java/src/test/java/com/tigerbeetle/AsyncRequestTest.java

```diff
@@ -24,7 +25,7 @@ public void testCreateAccountsRequestConstructor() {
             var batch = new AccountBatch(1);
             batch.add();
 
-            var request = AsyncRequest.createAccounts(client, batch);
+            var request = AsyncRequest.createAccounts(client, batch, null);
```

Comment:
Looking at the CompletableFuture API, it seems like it would be idiomatic to use overloading here, rather than a nullable parameter?

Reply:
Good point, we're using overloaded methods for the public API.
This case is just an internal class accepting a nullable parameter for the sake of simplicity.

However, even the public API can be considered non-idiomatic: the right call would be validating if the object is not null, and only accepting the default executor from the overloaded method which does not accept it as a parameter.

https://github.com/tigerbeetle/tigerbeetle/blob/c23944e7eb1671d70837c4a559bedc303bd3d74f/src/clients/java/src/main/java/com/tigerbeetle/Client.java#L138-L140